### ISSUE TITLE
Crash when opening files browser

### ISSUE THREAD

**Issue by KyotoFox** [merlos/iOS-Open-GPX-Tracker#271]
The app crashes consistently for me when I tap the "folder" icon in the top left. 
The view _does_ show up, and shows one GPX file that I think I tried to import previously, but after the view has fully animated in, the app crashes with this error in the console:
```
OpenGpxTracker/GPXFileInfo.swift:23: Fatal error: 'try!' expression unexpectedly raised an error: Error Domain=NSCocoaErrorDomain Code=257 "The file “ut-no_alsvik---skrussfjellet---nordland.gpx” couldn’t be opened because you don’t have permission to view it." UserInfo={NSURL=file:///private/var/mobile/Library/Mobile%20Documents/com~apple~CloudDocs/Kart/ut-no_alsvik---skrussfjellet---nordland.gpx, NSFilePath=/private/var/mobile/Library/Mobile Documents/com~apple~CloudDocs/Kart/ut-no_alsvik---skrussfjellet---nordland.gpx, NSUnderlyingError=0x301675200 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}
```

I do still have a file in iCloud Drive named `Kart/ut-no_alsvik---skrussfjellet---nordland.gpx`, but for some reason the app doesn't have access to it?

I haven't tried reinstalling the app yet. That might fix it, but I thought I would report the bug so that it can get fixed! 
When the app is in this state, I can save files, but they "go into the void". They don't show up in any folders in the Files app. The only way I've been able to get files out in this state is by sharing them through the share sheet instead.

I'm not that versed in Swift, but could it be that the issue is that `.modifiedDate` is called _after_ `customFolderURL.stopAccessingSecurityScopedResource()` in this location?
https://github.com/merlos/iOS-Open-GPX-Tracker/blob/fbd582ed6af1ca637097ede12078c8ef2be88c60/OpenGpxTracker/GPXFileManager.swift#L38-L45

**Comment by merlos** [comment]
I was able to reproduce the issue (or maybe related) by assigning a custom folder and then renaming the folder from the file browser.

**Comment by merlos** [comment]
There were two problems:

 * In `GPXFileInfo` when getting `fileSize` and `modifiedDate`, when the filesystem returned error getting the properties it was not properly handled. (mostly fixed in [2e39c29c6518632515654d2bc1065fe981f57dcd](https://github.com/merlos/iOS-Open-GPX-Tracker/commit/2e39c29c6518632515654d2bc1065fe981f57dcd))
 
 * Then when changing the folder name, to use a file, it was required the use of `startAccessingSecurityScopedResource`. In the original code it was used only while getting list of files, but not when displaying the files in the table or when opening the files. 
The easiest solution was to `startAccessingSecurityScopedResource` and stop on deinit of the view controller (~ fixed in https://github.com/merlos/iOS-Open-GPX-Tracker/commit/ace3fb796d6c7335f150da7c74371721543b85c1)
